frontend/include/chpl/framework/error-classes-list.h

```diff
@@ -85,3 +85,61 @@ ERROR_CLASS(ProcTypeUnannotatedFormal,
 ERROR_CLASS(ProcDefExplicitAnonFormal,
     const uast::Function*,
     const uast::Formal*)
+
+/* parser errors */
+PARSER_ERROR_CLASS(UnknownPragma, std::string)
+PARSER_ERROR_CLASS(TypeCannotImplementInterface, std::string)
+PARSER_ERROR_CLASS(PragmasBeforeDeprecation)
+PARSER_ERROR_CLASS(CannotAttachPragmas)
+PARSER_ERROR_CLASS(PrimCallNamedArgs)
+PARSER_ERROR_CLASS(PrimCallNoStrLiteral)
+PARSER_ERROR_CLASS(UnknownPrimitive)
+PARSER_ERROR_CLASS(SecondaryTypeMethodNoType)
+PARSER_ERROR_CLASS(ThisIntentNotMethod)
+PARSER_ERROR_CLASS(InvalidIndexExpr)
+PARSER_ERROR_CLASS(ClassExportExtern)
+PARSER_ERROR_CLASS(UnionExport)
+PARSER_ERROR_CLASS(RecordInheritanceNotSupported)
+PARSER_ERROR_CLASS(UnionInheritanceNotAllowed)
+PARSER_ERROR_CLASS(MultipleInheritance)
+PARSER_ERROR_CLASS(InheritInvalidExpr)
+PARSER_ERROR_CLASS(InvalidNumericLiteral, std::string, std::string)
+PARSER_ERROR_CLASS(MultipleExternalRenaming)
+PARSER_ERROR_CLASS(ReduceIntentNoIdent)
+PARSER_WARNING_CLASS(PreIncDecOp, std::string, std::string)
+PARSER_ERROR_CLASS(NewWithoutArgs)
+PARSER_ERROR_CLASS(DotAfterNew)
+PARSER_ERROR_CLASS(InvalidNewForm)
+PARSER_ERROR_CLASS(InvalidIterandExpr)
+PARSER_ERROR_CLASS(UseImportNeedsModule, const bool)
```

Comment:
IMO we don't need the const qualifiers on error info parameters. If we wanted to be very careful, we could enforce const-ness by making the `std::tuple` in the error be `const` instead.